### ISSUE TITLE
pdf thumbnail generation failed: "Unable to set image alpha channel"

### ISSUE THREAD

**Issue by elbidone** [meumobi/sitebuilder#389]
Latest pdf thumbs generation failed (tested on elefante). Below latest examples from institutoayrtonsenna.org.br and cimentonacional.com.br. They spend both almost 200 sec for processing and failed. We should investigate why they failed and if we should alter imageMagick configuration to succeed.

I use following command to find these issues
`$ cat log/sitebuilder.log-20160626 | grep "thumbnail generation failed" | wc -l`

Here a link from stackoverflow to investigate [http://stackoverflow.com/questions/10805122/imagemagick-setimagealphachannel-not-working-php](http://stackoverflow.com/questions/10805122/imagemagick-setimagealphachannel-not-working-php)

```bash
[2016-06-24 17:29:39] sitebuilder.ERROR: thumbnail generation failed {"item_id":"576d3bdd9a645d371325b497
","medium_url":"http://www.institutoayrtonsenna.org.br/wp-content/uploads/2015/07/Instituto-Ayrton-Senna-Rela
torio-Anual-de-Resultados-2014.pdf","type":"application/pdf","message":"Unable to set image alpha channel","e
xception":"[object] (ImagickException(code: 1): Unable to set image alpha channel at /home/meumobi/PROJECTS/m
eumobi.com/releases/20160614185137/sitebuilder/lib/meumobi/sitebuilder/services/PdfThumbnailer.php:37)","comp
onent":"media_thumbnailer"}
[2016-06-24 17:29:39] sitebuilder.NOTICE: thumbnail generation failed {"item_id":"576d3bdd9a645d371325b49
7","medium_url":"http://www.institutoayrtonsenna.org.br/wp-content/uploads/2015/07/Instituto-Ayrton-Senna-Rel
atorio-Anual-de-Resultados-2014.pdf","type":"application/pdf","error":null,"processing_time":181.10546708107,
"component":"media_thumbnailer"}
[2016-06-27 16:34:37] sitebuilder.ERROR: thumbnail generation failed {"item_id":"577138a09a645d400de87966","medium_url":"http://www.cimentonacional.com.br/wp-content/uploads/2015/02/folderinstitucional.pdf","type":"application/pdf","message":"Unable to set image alpha channel","exception":"[object] (ImagickException(code: 1): Unable to set image alpha channel at /home/meumobi/PROJECTS/meumobi.com/releases/20160614185137/sitebuilder/lib/meumobi/sitebuilder/services/PdfThumbnailer.php:37)","component":"media_thumbnailer"}
[2016-06-27 16:34:37] sitebuilder.NOTICE: thumbnail generation failed {"item_id":"577138a09a645d400de87966","medium_url":"http://www.cimentonacional.com.br/wp-content/uploads/2015/02/folderinstitucional.pdf","type":"application/pdf","error":null,"processing_time":214.03705596924,"component":"media_thumbnailer"}
```

**Comment by elbidone** [comment]
Another example that occurs this morning:

```bash
[2016-07-13 15:49:37] sitebuilder.DEBUG: file read {"medium_url":"http://www.sandoz.com.br/cs/www.sandoz.com.br/download/Guia_Bilingue.pdf","component":"media_thumbnailer"}
[2016-07-13 15:49:37] sitebuilder.ERROR: thumbnail generation failed {"item_id":"5786462d9a645d412035cc63","medium_url":"http://www.sandoz.com.br/cs/www.sandoz.com.br/download/Guia_Bilingue.pdf","type":"application/pdf","message":"Unable to set image alpha channel","exception":"[object] (ImagickException(code: 1): Unable to set image alpha channel at /home/meumobi/PROJECTS/meumobi.com/releases/20160614185137/sitebuilder/lib/meumobi/sitebuilder/services/PdfThumbnailer.php:37)","component":"media_thumbnailer"}
[2016-07-13 15:49:37] sitebuilder.NOTICE: thumbnail generation failed {"item_id":"5786462d9a645d412035cc63","medium_url":"http://www.sandoz.com.br/cs/www.sandoz.com.br/download/Guia_Bilingue.pdf","type":"application/pdf","error":null,"processing_time":154.29669213295,"component":"media_thumbnailer"}
```

**Comment by juliogreff** [comment]
"Unable to set image alpha channel" has been fixed (commit coming today) and will be live in the next deploy.

It takes time because it has to download the file, and when it's downloaded we start processing the file. That's when we get the error.

**Comment by elbidone** [comment]
I've done tests this morning and it works when calling `$image->setImageAlphaChannel(11)` inside a conditional `if ($image->getImageAlphaChannel()) {...}`

```php
    // Image has transparency
    if ($image->getImageAlphaChannel()) {
      $image->setImageAlphaChannel(11); // Imagick::ALPHACHANNEL_REMOVE
    }
```

I've setted it on ELEFANTE and it works well for pdf listed above

**Comment by juliogreff** [comment]
Yeah, that's the fix I applied.

**Comment by malord** [comment]
Hi. The fix appears to be missing from version 1.5.5. Manually applying it to my pdf-image-generator.php has just fixed the issue for me (Ubuntu 14.04).
